src/Periphery/LiFiDEXAggregator.sol

```diff
@@ -732,6 +748,163 @@ contract LiFiDEXAggregator is WithdrawablePeriphery {
         uniswapV3SwapCallback(amount0Delta, amount1Delta, data);
     }
 
+    /// @notice Performs a swap through iZiSwap V3 pools
+    /// @dev This function handles both X to Y and Y to X swaps through iZiSwap V3 pools
+    /// @param stream [pool, direction, recipient]
+    /// @param from Where to take liquidity for swap
+    /// @param tokenIn Input token
+    /// @param amountIn Amount of tokenIn to take for swap
+    function swapIzumiV3(
+        uint256 stream,
+        address from,
+        address tokenIn,
+        uint256 amountIn
+    ) private {
+        address pool = stream.readAddress();
+        uint8 direction = stream.readUint8(); // 0 = Y2X, 1 = X2Y
+        address recipient = stream.readAddress();
+        if (
+            pool == address(0) ||
+            pool == IMPOSSIBLE_POOL_ADDRESS ||
+            recipient == address(0) ||
+            amountIn > type(uint128).max
+        ) revert InvalidCallData();
+
+        if (from == msg.sender) {
+            IERC20(tokenIn).safeTransferFrom(
+                msg.sender,
+                address(this),
+                amountIn
+            );
+        }
+
+        lastCalledPool = pool;
+
+        if (direction == DIRECTION_TOKEN0_TO_TOKEN1) {
+            IiZiSwapPool(pool).swapX2Y(
+                recipient,
+                uint128(amountIn),
+                IZUMI_LEFT_MOST_PT + 1,
+                abi.encode(tokenIn)
+            );
+        } else {
+            IiZiSwapPool(pool).swapY2X(
+                recipient,
+                uint128(amountIn),
+                IZUMI_RIGHT_MOST_PT - 1,
+                abi.encode(tokenIn)
+            );
+        }
+
+        if (lastCalledPool != IMPOSSIBLE_POOL_ADDRESS) {
```

Comment:
here I initially wanted to move this up (before the swaps) but then realized it would change the logic. 
Please add comments to make it clear why certain things are done. This is not clear from just reading the code.